### PR TITLE
Normalize ASTs in ColumnsDescription

### PR DESCRIPTION
**Pull request by CurtizJ**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix (user-visible misbehaviour in official stable or prestable release)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix possible `Table columns structure in ZooKeeper is different from local table structure` exception while recreating or creating new replicas of `ReplicatedMergeTree`, when one of table columns have default expressions with case-insensitive functions.

### CODE REVIEW

File: src/Storages/tests/gtest_columns_description_normalize.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+#include <gtest/gtest.h>
+
+using namespace DB;
+
+TEST(ColumnsDescription, Normalize)
+{
+    constexpr auto columns = R"(columns format version: 1
+3 columns:
+`a` UInt32
+`b` String	DEFAULT	If(a = 0, \'true\', \'false\')
```
Comment (vdimir) [inline src/Storages/tests/gtest_columns_description_normalize.cpp:13]: Why do we need escape `'` here?

File: src/Storages/tests/gtest_columns_description_normalize.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+`c` String	DEFAULT	cAsT(a, 'String')
+)";
```
Comment (vdimir) [inline src/Storages/tests/gtest_columns_description_normalize.cpp:15]: I'd prefer to format such strings like this, but it's up to you.
```suggestion
    constexpr auto columns = "columns format version: 1\n"
                             "3 columns:\n"
                             "`a` UInt32\n"
                             "`b` String	DEFAULT	If(a = 0, \'true\', \'false\')\n"
                             "`c` String	DEFAULT	cAsT(a, 'String')\n";
```

File: src/Storages/ColumnsDescription.cpp [hunk 3]
```diff
@@ -201,6 +202,13 @@ void ColumnsDescription::add(ColumnDescription column, const String & after_colu
         throw Exception("Cannot add column " + column.name + ": column with this name already exists",
             ErrorCodes::ILLEGAL_COLUMN);
 
+    /// Normalize ASTs to be compatible with InterpreterCreateQuery.
+    FunctionNameNormalizer normalizer;
+    if (column.default_desc.expression)
+        normalizer.visit(column.default_desc.expression.get());
+    if (column.ttl)
+        normalizer.visit(column.ttl.get());
```
Comment (vdimir) [inline src/Storages/ColumnsDescription.cpp:210]:
```suggestion
    if (column.default_desc.expression)
        FunctionNameNormalizer::visit(column.default_desc.expression.get());
    if (column.ttl)
        FunctionNameNormalizer::visit(column.ttl.get());
```

**Review by vdimir — APPROVED** [review]

**Comment by CurtizJ** [comment]
The same integration tests are broken in master.

**Pull request merged**